cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -307,14 +307,18 @@ def _add_head_eni(self):
             group_set=head_eni_group_set,
         )
 
-        # Create and associate EIP to Head Node
-        if self.config.head_node.networking.elastic_ip is True:
-            head_eip = ec2.CfnEIP(scope=self, id="HeadNodeEIP", domain="vpc")
-
+        elastic_ip = self.config.head_node.networking.elastic_ip
+        if elastic_ip:
+            # Create and associate EIP to Head Node
+            if elastic_ip is True:
+                allocation_id = ec2.CfnEIP(scope=self, id="HeadNodeEIP", domain="vpc").attr_allocation_id
+            # Attach existing EIP
+            elif elastic_ip:
```

Comment:
this can be a simple else, because you're already testing `if elastic_ip` in row 311

Reply:
Done